iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/functional_interface.inl

```diff
@@ -37,7 +37,10 @@ inline void Expect<Derived>::expect(const StringType& msg) const noexcept
 
     if (!(*static_cast<const Derived*>(this)))
     {
-        std::cout << msg << std::endl;
+        // it is possible that expect is called inside a signal handler therefore we
+        // use write
+        auto result = write(STDERR_FILENO, &msg[0], strlen(&msg[0]));
+        IOX_DISCARD_RESULT(result);
         Ensures(false);
```

Comment:
@elBoberido you are right `Require` in `requires.cpp` has to issues
* `std::cerr` which we can handle with `write`
* a call to the `errorHandler`

Now the question is, should the `errorHandler` be signal safe. I think this is nearly impossible. The other option would be to not use `expect` in a signal handler then I guess.

Reply:
So `expect` is not signal handler safe and uses `LogFatal` as output channel. But since the logger has a dependency to `convert` which has a dependency to `string` which has a dependency to `optional` and `expected` I had to move the output logic into a cpp file otherwise I would have a cyclic dependency.

Reply:
Yes, I think we cannot make the `errorHandler` signals safe. After the refactoring one could replace it with a custom implementation and a signal safe limitation would also limit the error reporting quite a lot. I think we have to make clear that neither the `errorHandler` nor the `Logger` should be used from a signal handler.